src/db/sysdb_search.c

```diff
@@ -545,8 +550,19 @@ static char *enum_filter(TALLOC_CTX *mem_ctx,
         filter = talloc_asprintf(tmp_ctx, "(&%s", base_filter);
 
         if (filter != NULL && attr != NULL && attr_filter != NULL) {
-            filter = talloc_asprintf_append(filter, "(%s=%s)",
-                                            attr, attr_filter);
+            if (domain == NULL || !IS_SYSDB_NAME(attr)) {
+                value = attr_filter;
+            } else {
+                value = sss_create_internal_fqname(tmp_ctx, attr_filter, domain);
+                if (value == NULL) {
+                    filter = NULL;
+                }
+            }
+
+            if (filter != NULL) {
+                filter = talloc_asprintf_append(filter, "(%s=%s)",
+                                                attr, value);
+            }
```

Comment:
What other use cases, besides infopipe.*.ListByName, affects this change?

Reply:
Which cases it affects depends on whether the `domain` parameter is `NULL`.
This function is used from 4 different locations but only 2 pass a non-`NULL` `domain`. Those two functions are used in the ListByName/ListByAttr (the same one) and GroupByName callchains because it looks to me that in this last case it is needed. If I am wrong, just let me know and it can easily be fixed.

Reply:
`plugins/cache_req_user_by_filter.c`  --  is it only used by `ifp`?

Reply:
Why do you think `SSS_NSS_GETGRENT` is affected?

Reply:
I don't remember the details, but At some moment, tracing back the function calls, I think I reached to `SSS_NSS_GETGRENT`.

Reply:
> I don't remember the details, but At some moment, tracing back the function calls, I think I reached to `SSS_NSS_GETGRENT`.

`sysdb_enumgrent_with_views()` executes `sysdb_enumgrent_filter_with_views(... NULL ...)` so I think shouldn't be affected.

So this is all about IFP?

Reply:
It seems, as I initially thought.